src/Design/03_InitialPoC_trans_ffi.purs

```diff
@@ -0,0 +1,18 @@
+module FFI.PSPS.Design.InitialPoC.Trans where
+
+import PSPS (App, Foreign, Literal, Prim'Int, kind Expr, kind FFIName)
+import Type.Data.Peano (class ProductInt, class SumInt)
+
+class FFI_PSPS_Design_InitialPoC_Trans_Eval (expr :: Expr) (result :: Expr) | expr -> result
```

Comment:
Jumping to the FFI, here is where the magic happens. Because the PS compiler is bug-free, we _know_ that `FFI_PSPS_Design_InitialPoC_Trans_Eval (App (App (Foreign Foreign'intMul) a') b')` above will result in `FFI_PSPS_Design_InitialPoC_Trans_Eval (App (App (Foreign Foreign'intAdd) (Literal (Prim'Int a))) (Literal (Prim'Int b))) (Literal (Prim'Int c))` below. How could it be anything else? In `corefn`, Purescript _only_ ever passes us programs that compile. So if both arguments to `FFI_PSPS_Design_InitialPoC_Trans_Eval` are fully evaluated, and if they passed the purescript compiler, they MUST be int literals. To grok this, imagine: what if I wrote `intAdd "foo" "bar"`. What would the corefn be? The answer is "nothing". `corefn` would never even compile `intAdd "foo" "bar"` or `intAdd true false`. It will _only_ allow `intAdd` over integers, ie `intAdd 42 43`. So, in the FFI file, we can safely assume that if everything in the non-FFI files was compiled without error, then we _must_ unpack two literal integers.